### PR TITLE
Complete DI for business controller class

### PR DESCRIPTION
**Pull request by bdukes**
## Summary
This PR relies on #5618. It adds the rest of the support for dependency injection for business controller class.

### CODE REVIEW

**Comment by mitchelsellers** [comment]
@bdukes Looks like we have a build failure don't his one as well

**Comment by bdukes** [comment]
tests are passing now

**Review by mitchelsellers — APPROVED** [review]

File: DNN Platform/Library/Services/Syndication/RssHandler.cs [hunk 1]
```diff
@@ -19,6 +19,13 @@ namespace DotNetNuke.Services.Syndication
     /// <summary>An HTTP handler for serving an RSS feed.</summary>
     public class RssHandler : SyndicationHandlerBase
     {
+        private readonly ISearchController searchController;
+
+        public RssHandler(ISearchController searchController)
```
Comment (valadas) [inline DNN Platform/Library/Services/Syndication/RssHandler.cs:24]: I think we need to add an obsolete parameterless constructor for this one

File: DNN Platform/Library/UI/Modules/Html5/Html5HostControl.cs [hunk 2]
```diff
@@ -20,15 +20,26 @@ namespace DotNetNuke.UI.Modules.Html5
...
+        [Obsolete("Deprecated in DotNetNuke 10.0.0. Please use overload with IBusinessControllerProvider. Scheduled removal in v12.0.0.")]
         public Html5HostControl(string html5File)
         {
             this.html5File = html5File;
         }
 
+        /// <summary>Initializes a new instance of the <see cref="Html5HostControl"/> class.</summary>
+        /// <param name="html5File">The path to the HTML file.</param>
+        /// <param name="businessControllerProvider">The business controller provider.</param>
+        public Html5HostControl(string html5File, IBusinessControllerProvider businessControllerProvider)
```
Comment (valadas) [inline DNN Platform/Library/UI/Modules/Html5/Html5HostControl.cs:37]: I believe the old constructor need to be updated to call this one

File: DNN Platform/Library/UI/Modules/Html5/Html5ModuleControlFactory.cs [hunk 3]
```diff
@@ -1,31 +1,48 @@
...
+        public Html5ModuleControlFactory()
+            : this(null)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="Html5ModuleControlFactory"/> class.</summary>
+        /// <param name="businessControllerProvider">The business controller provider.</param>
+        public Html5ModuleControlFactory(IBusinessControllerProvider businessControllerProvider)
+        {
+            this.businessControllerProvider = businessControllerProvider;
```
Comment (valadas) [inline DNN Platform/Library/UI/Modules/Html5/Html5ModuleControlFactory.cs:27]: I think we need a ternary for a fallback on this one

**Review by valadas — COMMENTED** [review]
wow, that is a big one, I left a couple of comments

File: DNN Platform/Library/Obsolete/ModuleController.cs [hunk 4]
```diff
@@ -1,144 +1,198 @@
...
-            }
-
-            return settings;
-        }
-    }
-}
+    using Microsoft.Extensions.DependencyInjection;
+
+    /// <content>The deprecated methods for ModuleController.</content>
+    public partial class ModuleController
```
[the remark this answers is not in the document]
Comment (bdukes) [reply DNN Platform/Library/Obsolete/ModuleController.cs]: This is a partial class for the obsolete methods of `ModuleController`, the class itself isn't obsolete.

**Review by valadas — APPROVED** [review]
Awesome!

**Pull request merged**